src/embeddings/embeddings_calculator.cc

```diff
@@ -138,52 +137,75 @@ class EmbeddingsCalculator : public CalculatorBase {
                 SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
                 embeddingsInputMap[embeddingsInputName] = it->second;
             }
-
             size_t max_context_length = 512;  // default allowed input length. Otherwise, it will be read from model rt_info>config>max_position_embeddings in the model.xml file
             try {
                 max_context_length = embeddings_session->getModelConfig().at("config").as<ov::AnyMap>().at("max_position_embeddings").as<size_t>();
             } catch (...) {
                 SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Can not read config->max_position_embeddings from model rt_info. Using default value {}", max_context_length);
             }
-
-            RET_CHECK(tokenizerOutputMap["input_ids"].get_shape().size() == 2);
-            size_t input_ids_size = tokenizerOutputMap["input_ids"].get_shape()[1];
+            RET_CHECK(tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape().size() == 2);
+            size_t input_ids_size = tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape()[1];
             if (input_ids_size > max_context_length) {
                 SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Input size {} exceeds max_context_length {}", input_ids_size, max_context_length);
                 return absl::InvalidArgumentError(absl::StrCat("Input length ", input_ids_size, " longer than allowed ", max_context_length));
             }
-
-            embeddingsOutputMap = embeddings_session->infer(embeddingsInputMap);
-            if (embeddingsOutputMap.empty()) {
+            // creating output to avoid copying big tensor
+            ov::Shape outShape = embeddingsInputMap.at(EMBEDDINGS_MODEL_INPUT_IDS_NAME).get_shape();
+            bool foundMatchinEmbeddingOutput{false};
+            std::string outputNameToSet;
+            for (auto& name : embeddings_session->getOutputNames()) {
+                ov::PartialShape outPShape = embeddings_session->getOutputShape(name);
+                if (outPShape.size() != 3)
+                    continue;
+                try {
+                    outShape.emplace_back(outPShape[2].get_length());
+                    foundMatchinEmbeddingOutput = true;
+                    outputNameToSet = name;
+                    break;
+                } catch (std::exception& e) {
+                    LOG(ERROR) << "Failed to get 3rd dimension of output" << outputNameToSet;
```

Comment:
mixture of LOG and SPDLOG loggers

Reply:
SPDLOG in calculators does not work at main branch.